tox.ini

```diff
@@ -11,11 +11,12 @@ setenv =
     PYTHONHASHSEED = 42
 commands =
     python setup.py build_ext --inplace
-    py27,py34,py35: nosetests -v zarr
-    py36: nosetests -v --with-coverage --cover-erase --cover-min-percentage=100 --cover-package=zarr --with-doctest --doctest-options=+NORMALIZE_WHITESPACE zarr
+    py27: nosetests -v --with-coverage --cover-erase --cover-package=zarr zarr
+    py34,py35,py36: nosetests -v --with-coverage --cover-erase --cover-package=zarr --with-doctest --doctest-options=+NORMALIZE_WHITESPACE zarr
```

Comment:
Are the `dict` ordering issues related to Python's salting of hashes? If so, I would have thought setting `PYTHONHASHSEED` to a fixed value would solve this issue. Though a more robust solution would probably be to use `OrderedDict`s. That said, given you have set `PYTHONHASHSEED` and it was working before, I'm a bit curious to know if there is more to the story.